chain-abci/src/app/mod.rs

```diff
@@ -297,6 +300,32 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
                     validator.set_power(i64::from(*new_power));
                     validator.set_pub_key(self.validator_pubkeys[&address].clone());
                     validators.push(validator);
+
+                    let last_state = self
+                        .last_state
+                        .as_mut()
+                        .expect("Last app state not found, init chain was not called");
+
+                    let validator_liveness = &mut last_state.validator_liveness;
+
+                    let validator_address: TendermintValidatorAddress =
+                        into_tendermint_validator_pub_key(&self.validator_pubkeys[&address]).into();
+
+                    if i64::from(*new_power) == 0
```

Comment:
this could be named: 
```
let new_vote_power = TendermintVotePower::from(*new_power);
if new_vote_power == 0 {
...
} else if new_vote_power != 0 && !validator_liveness.contains_key(&validator_address) {
...
}
```
```